### PR TITLE
Refactor and update blockaid banner alert

### LINKED ISSUE #22624: [Refactor]: Update BannerAlert "NotApplicable" logic; minor clean up; update storybook support
### Describe the bug

Before working on more updates to the BlockaidBannerAlert

See PR for more details

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

1. open storybook `yarn storybook`
2. go to http://localhost:6006/?path=/docs/components-app-securityproviderbanneralert-blockaidbanneralert--docs

### Error messages or log output

_No response_

### Version

11.7.3

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

minor cleanup before more updates on BlockaidBannerAlert and SecurityProviderBannerAlert
- update domain prop to include txData?.origin
- display "Provided by" even on failed tx
- update storybook controls to redisplay dropdown and without "NotApplicable" types
- alphabetize params
- rm redundant getDisclosureTitle function

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/22624

## **Manual testing steps**

1. run storybook yarn storybook
2. go to http://localhost:6006/?path=/docs/components-app-securityproviderbanneralert-blockaidbanneralert--docs

### **Before**

Screenshot 2024-01-22 at 8 32 28 AM

### **After**

Screenshot 2024-01-22 at 10 51 05 AM
Screenshot 2024-01-22 at 10 51 12 AM

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js [hunk 1]
```diff
@@ -88,41 +88,41 @@ function BlockaidBannerAlert({ txData, ...props }) {
...
-    resultType,
-    reproduce: JSON.stringify(features),
-  };
+  const reportUrl = (() => {
+    const reportData = {
+      blockNumber: block,
+      blockaidVersion: BlockaidPackage.version,
+      chain: NETWORK_TO_NAME_MAP[chainId],
+      classification: reason,
+      domain: origin ?? msgParams?.origin,
```
Comment (digiwand) [inline ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:104]: Q: should this be 
```
domain: origin ?? msgParams?.origin ?? txParams?.origin
```
? 
@jpuri @segun

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js [hunk 2]
```diff
@@ -88,41 +88,41 @@ function BlockaidBannerAlert({ txData, ...props }) {
...
+      jsonRpcMethod: type,
+      jsonRpcParams: JSON.stringify(txParams ?? msgParams),
+      resultType,
+      reproduce: JSON.stringify(features),
+    };
```
Comment (digiwand) [inline ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:109]: no key/values updated here. only alphabetized

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js (hunk 1 above)
Comment (jpuri) [reply ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:106]: yep that would be better @digiwand

**Review by jpuri — DISMISSED** [review]

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js (hunk 1 above)
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js]: thanks @jpuri! fixed [046c27c](https://github.com/MetaMask/metamask-extension/pull/22625/commits/046c27cd2c28863c5d6ab0fcd829721b80a64e2f)

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js [hunk 3]
```diff
@@ -88,41 +88,41 @@ function BlockaidBannerAlert({ txData, ...props }) {
...
+      blockNumber: block,
+      blockaidVersion: BlockaidPackage.version,
+      chain: NETWORK_TO_NAME_MAP[chainId],
+      classification: reason,
+      domain: origin ?? msgParams?.origin ?? txParams?.origin,
+      jsonRpcMethod: type,
+      jsonRpcParams: JSON.stringify(txParams ?? msgParams),
+      resultType,
+      reproduce: JSON.stringify(features),
+    };
```
Comment (digiwand) [inline ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:109]: the only key/value prop updated here is domain. all others were reordered alphabetically

**Review by jpuri — DISMISSED** [review]

**Review by segun — DISMISSED** [review]

**Review by segun — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Pull request merged**